app/assets/scripts/map.js

```diff
@@ -150,59 +159,76 @@ function create_marker(resource){
         $("#map").show();
         $("#resource-info").hide();
 
-        var contentDiv = document.createElement('div');
-        var resName = document.createElement('strong');
-        $(resName).html(resource.name);
-        var br = document.createElement('br');
-
-        // information pane to replace map
-        // TODO: refactor how the page elements are generated
-        var moreLink = document.createElement('p');
-        moreLink.innerHTML = 'More Info';
-        moreLink.setAttribute('class', 'more-info')
-        moreLink.addEventListener('click', function() {
+        // Show marker info bubble for clicked marker
+        var markerTemplate = $("#marker-template").html();
+        var compiledMarkerTemplate = Handlebars.compile(markerTemplate);
+        var context = {
+          name: resource.name,
+          address: resource.address,
+        };
+        var markerInfo = compiledMarkerTemplate(context);
+
+        if (infowindow) {
+          infowindow.close();
+        }
+        infowindow = new google.maps.InfoWindow({
+          content: markerInfo
+        });
+        infowindow.open(map, markerToAdd);
+        map.setCenter(markerToAdd.getPosition());
+        map.setZoom(focusZoom);
+
+        // Marker to detailed resource information view
+        $(".more-info").click(function() {
           // get descriptor information
           $.get('get-associations/' + resource.id).done(function(associations) {
             $("#map").hide();
             $("#resource-info").empty();
             $("#resource-info").show();
 
-            var backButton = document.createElement('button');
-            backButton.innerHTML = 'Back';
-            backButton.setAttribute('class', 'ui button');
-            backButton.addEventListener('click', function() {
+            var associationObject = JSON.parse(associations);
```

Comment:
`JSON.parse` turns the JSON returned from the get request on line 184 into an JS object - I feel like the documentation is just what the method call itself does so is it necessary?